src/transformers/models/gemma2/modeling_gemma2.py

```diff
@@ -747,8 +747,24 @@ def forward(
             inputs_embeds = self.embed_tokens(input_ids)
 
         if cache_position is None:
+            if past_key_values is not None and not past_key_values.empty:
+                logger.warning(
+                    "You are calling the model with non-empty `past_key_values` but didn't pass `cache_position`. ",
+                    "This will results in incorrect logits. Please pass `cache_position` that indicates indicates "
+                    "input's positions in the sequence.",
+                )
             cache_position = torch.arange(0, inputs_embeds.shape[1], device=inputs_embeds.device)
```

Comment:
```suggestion
        if cache_position is None:
            if past_key_values is None:
                cache_position = torch.arange(0, inputs_embeds.shape[1], device=inputs_embeds.device)
            else:
                raise ValueError("When `past_key_values` is passed, `cache_position` must be too")
```

Since we're moving away from `cache.get_seq_length()`, we can't infer `cache_position` when a cache is passed. Let's raise an exception -- users often ignore warnings :)

Reply:
Okey, let it be an error then